### PR TITLE
Update native tutorial

### PR DESCRIPTION
**Pull request by mit-mit**
Update the first tour under command-line and server apps to focus more on the tooling/workflow: creating an app, running it in the VM, debugging with DevTools, and AOT-compiling. This works as the following two tutorials focus more on the actual content for 1. command-line, and 2. client/server-apps.

Ported from https://github.com/dart-lang/site-www/pull/1460, and updated to reflect new IA.

### CODE REVIEW

**Comment by mit-mit** [comment]
This probably needs some copy-editing; feel free to push those directly to the PR.

File: src/_tutorials/server/get-started.md [hunk 1]
```diff
@@ -1,263 +1,237 @@
 ---
-title: Get started with server-side Dart
-description: Get Dart and run a Dart app.
+title: "Get started: Dart native apps"
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: Dart native or Dart Native?

/cc @messi01

Comment (messi01) [reply src/_tutorials/server/get-started.md]: @kwalrath what's the context for this? Where is this tutorial on the current dartlang site? Have we historically used "Dart native"?

**Comment by kwalrath** [comment]
Staged:
https://kw-www-dartlang-1.firebaseapp.com

File: src/_tutorials/server/get-started.md [hunk 2]
```diff
@@ -1,263 +1,237 @@
...
 ---
 
-<div class="mini-toc" markdown="1">
-  <h4>What's the point?</h4>
-
-  * The Dart SDK has development tools and libraries.
-  * Use an IDE or a code editor to create your app.
-  * All Dart apps have a `main()` function.
-  * Dart supports top-level functions.
+Follow these steps to start using Dart to develop web apps. First you’ll play
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: web -> ???

File: src/_tutorials/server/get-started.md [hunk 3]
```diff
@@ -1,263 +1,237 @@
...
+with Dart in your browser, no download required. Then you’ll install Dart and
+run a small program with the Dart VM. Finally, you will AOT-compile that program
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: you will -> you'll

(https://developers.google.com/style/contractions)

File: src/_includes/server-tutorials.md [hunk 4]
```diff
@@ -4,7 +4,8 @@ and server apps that can run in the standalone Dart VM.
 <div class="card-grid">
   <div class="card">
     <h3><a href="/tutorials/server/get-started">Get started</a></h3>
```
Comment (kwalrath) [inline src/_includes/server-tutorials.md]: Get started with Dart Native?

Comment (kwalrath) [reply src/_includes/server-tutorials.md]: If we're going to use that terminology, then we should use it everywhere and introduce/mention it in the intro to this content.

File: src/_tutorials/server/get-started.md (hunk 1 above)
Comment (kwalrath) [reply src/_tutorials/server/get-started.md]: @messi01 I was just writing you email... I hope to get it to you within the hour!

File: src/_tutorials/server/get-started.md [hunk 5]
```diff
@@ -1,263 +1,237 @@
...
+title: "Get started: Dart native apps"
+description: Get Dart, run and compile a small app
+toc: true
 nextpage:
   url: /tutorials/server/cmdline
   title: Write command-line apps
 prevpage:
   url: /tutorials/server
-  title: Server-side Dart tutorials
+  title: Tutorials
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: I'm not sure about this nav change.

File: src/_tutorials/server/get-started.md (hunk 1 above)
Comment (kwalrath) [reply src/_tutorials/server/get-started.md]: @mit-mit, I thought Flutter counted as Dart native? I'm getting confused...

File: src/_articles/server/c-interop.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+---
+title: "C/C++ interop"
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply src/_articles/server/c-interop.md]: For the target market (native developers) C/C++ is a well understood shorthand for C and C++.

File: src/_tutorials/server/get-started.md (hunk 1 above)
Comment (mit-mit) [reply src/_tutorials/server/get-started.md]: > @mit-mit, I thought Flutter counted as Dart native? I'm getting confused...

Flutter apps are powered by Dart Native, that right. Can you clarify what the confusion is?

Comment (mit-mit) [reply src/_tutorials/server/get-started.md]: > I'd love some more context on this! Where is this tutorial on the current dartlang site?

This doesn't currently exists. Historically we have only had the Dart VM (the `dart` command). The two new things in this PR we are adding are Dart AOT (the `dart2aot` command), and debugging using Dart DevTools.

File: src/_articles/server/c-interop.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mit-mit) [reply src/_articles/server/c-interop.md]: Yes, C/C++ is very common short-hand. I get 80 mio.+ results here: https://www.google.com/search?q="c%2Fc%2B%2B"

But, from this comment from Stroustrup (the creator of C++), maybe we shouldn't use it:
http://www.stroustrup.com/bs_faq.html#C-slash

File: src/_includes/server-tutorials.md (hunk 4 above)
Comment (mit-mit) [reply src/_includes/server-tutorials.md]: Used 'Get started: command-line & server apps' to be consistent with /web/get-started (which has 'Get started: web apps')

File: src/_tutorials/server/get-started.md (hunk 5 above)
Comment (mit-mit) [reply src/_tutorials/server/get-started.md]: This link is the the tutorial that is in this PR. That is a general tutorial that teaches the tools used for all command-line & server apps, so having the title just say 'server-side' is wrong.

Comment (kwalrath) [reply src/_tutorials/server/get-started.md]: Right, but it's not the top-level tutorial page, either. (e.g. it doesn't cover web apps & Flutter)

File: src/_tutorials/server/get-started.md [hunk 7]
```diff
@@ -1,263 +1,237 @@
...
+## 8. Compile for production
 
-## Run a command-line app {#run-cmd-line}
+The steps above used `dart`, the Dart VM to run the app. The Dart VM is
+optimized for fast, incremental compilation to provide instant feedback
+duringdevelopment. Now that our small app is done, it's time to AOT compile our
+Dart code to optimized native machine code.
 
-Run the app, using either your IDE or the command line:
+Use the `dart2aot` tool to AOT-compile the program to machine code:
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: We aren't even putting `dart2aot` in the changelog. (See https://github.com/dart-lang/sdk/issues/36758.) It feels very wrong to put it into this tutorial, unless this page isn't linked to by anything.

Comment (mit-mit) [reply src/_tutorials/server/get-started.md]: I added a page for `dart2aot` here: https://github.com/dart-lang/site-www/pull/1500/files#diff-a05d7dd79fd05505679e3fa9f09c618e

Comment (mit-mit) [reply src/_tutorials/server/get-started.md]: > We aren't even putting dart2aot in the changelog

Send you a CL.

File: src/_tutorials/server/get-started.md (hunk 5 above)
Comment (mit-mit) [reply src/_tutorials/server/get-started.md]: Changed to 'Dart command-line and server tutorials'

File: src/_tutorials/server/get-started.md [hunk 8]
```diff
@@ -1,263 +1,237 @@
...
-  * The Dart SDK has development tools and libraries.
-  * Use an IDE or a code editor to create your app.
-  * All Dart apps have a `main()` function.
-  * Dart supports top-level functions.
+Follow these steps to start using Dart to develop command-line & server apps.
+First you’ll play with Dart in your browser, no download required. Then you’ll
+install Dart and run a small program with the Dart VM. Finally, you'll
+AOT-compile that program to native machine code.
+
+## 1. Play with a command-line app in DartPad
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: a command-line app -> Dart code

(We don't want to imply that DartPad uses the VM or supports dart:io.)

File: src/_tutorials/server/get-started.md [hunk 9]
```diff
@@ -1,263 +1,237 @@
...
+With DartPad you can experiment with the Dart language and APIs,
+no download necessary.
+
+For example, here's an embedded DartPad that lets you play with
+the code for a small Hello World program.
+Click run {% asset red-run.png alt="" %} to run the app;
+the console output appears beneath the code.
+Try editing the source code—perhaps you'd like to change the greeting
+to use another language. To get the full DartPad experience,
+which includes the web UI that the app produces,
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: Delete this line, since the program has no web UI.

File: src/_tutorials/server/get-started.md [hunk 10]
```diff
@@ -1,263 +1,237 @@
...
+</iframe>
+
+More information:
+
+* [DartPad documentation][]
+* [Dart language tour][]
+* [Dart library tour][]
+
+## 2. Install Dart
+
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: We need an intro here. The one in the web page seems good enough:

Once you’re ready to move beyond DartPad and develop real apps, you need the Dart SDK.

File: src/_tutorials/server/get-started.md [hunk 11]
```diff
@@ -1,263 +1,237 @@
...
+As you install, **note the path to the SDK.**
+You'll need it in step 3.
+
+<ul class="tabs__top-bar">
+  <li class="tab-link current" data-tab="tab-sdk-install-windows">Windows</li>
+  <li class="tab-link" data-tab="tab-sdk-install-linux">Linux</li>
+  <li class="tab-link" data-tab="tab-sdk-install-mac">Mac</li>
+</ul>
+
+<div id="tab-sdk-install-windows" class="tabs__content current" markdown="1">
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: This installation code should be includes shared by /web/get-started.

File: src/_tutorials/server/get-started.md [hunk 12]
```diff
@@ -1,263 +1,237 @@
...
+```terminal
+> mkdir quickstart
+> cd quickstart
+> stagehand console-full
+> pub get
+```
 
-Whether you use an IDE or the command line,
-you can create a basic app using standard Dart templates.
+This creates a minimal Dart app that has the following:
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: I wouldn't call it minimal. It has testing!

File: src/_tutorials/server/get-started.md [hunk 13]
```diff
@@ -1,263 +1,237 @@
...
 
-If you're using a [Dart-savvy IDE][ide], follow these instructions:
+* One main Dart source file: `bin/main.dart`. This contains one top-level
+  `main()` function. This is the entry point for your app.
+* One additional Dart file: `lib/cli.dart`. This contains the functionality of
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: this is called `lib/quickstart.dart` in my version.
quickstart -> cli above?

File: src/_tutorials/server/get-started.md [hunk 14]
```diff
@@ -1,263 +1,237 @@
...
+## 5. Run the app
 
-1. If the IDE doesn't automatically download the packages
-   that the app depends on, make the IDE run `pub get`.
+First we need to get the dependencies of the app using the
+[`pub`](/tools/pub/cmd) command:
 
-If you're using the command line, follow these instructions:
+```terminal
+> pub get
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: We already did this above. Delete from above?

File: src/_tutorials/server/get-started.md [hunk 15]
```diff
@@ -1,263 +1,237 @@
...
+    ```
 
-`test`
-: Contains test code for the hello_world app.
+More information:
+[Write Command-Line Apps](/tutorials/server/cmdline)
 
-`.dart_tool`
-: Support files used by `pub` and other Dart tools. You can ignore this.
+## 7. Use DevTools to debug the app
```
Comment (kwalrath) [inline src/_tutorials/server/get-started.md]: I think we should (1) put the relevant, foolproof instructions here (like we do for stagehand) and (2) improve the instructions. There's no help at that site for running the `dart` command, so I wasn't sure what to do.

Delete this step for now?

**Comment by mit-mit** [comment]
> This is at /tutorials/server/get-started, but the web one is at /web/get-started. OK by you if I move the web one to /tutorials/web/get-started?

Yes!

File: src/_tutorials/server/get-started.md (hunk 15 above)
Comment (mit-mit) [reply src/_tutorials/server/get-started.md]: Deleting, and filed https://github.com/dart-lang/site-www/issues/1522 to track re-adding

File: src/_tutorials/server/get-started.md (hunk 11 above)
Comment (mit-mit) [reply src/_tutorials/server/get-started.md]: Done!

**Comment by mit-mit** [comment]
@kwalrath I think I have incorporated all the great feedback!

**Comment by kwalrath** [comment]
OK, copy edit is done. We gloss over many things, though — in particular setting paths. It'd be good to get someone new to Dart to try this out. Or, at a minimum, to set expectations somehow.

File: src/_tutorials/server/cmdline.md [hunk 16]
```diff
@@ -6,7 +6,7 @@ nextpage:
   title: "Write HTTP clients & servers"
 prevpage:
   url: /tutorials/server/get-started
-  title: Get started with the Dart VM
+  title: Get started with Dart Native
```
Comment (kwalrath) [inline src/_tutorials/server/cmdline.md]: -> Get started: command-line and server apps

**Comment by mit-mit** [comment]
Do you think we land land this, and iterate later?

**Pull request merged**